tls_codec/src/primitives.rs

```diff
@@ -372,48 +372,39 @@ impl<T> SerializeBytes for PhantomData<T> {
     }
 }
 
-#[cfg(feature = "std")]
-pub mod box_impls {
-    use crate::{Deserialize, DeserializeBytes, Error, Serialize, SerializeBytes, Size};
-
-    use std::{
-        boxed::Box,
-        io::{Read, Write},
-        vec::Vec,
-    };
-
-    impl<T: Size> Size for Box<T> {
-        #[inline(always)]
-        fn tls_serialized_len(&self) -> usize {
-            self.as_ref().tls_serialized_len()
-        }
+impl<T: Size> Size for Box<T> {
+    #[inline(always)]
+    fn tls_serialized_len(&self) -> usize {
+        self.as_ref().tls_serialized_len()
     }
+}
 
-    impl<T: Serialize> Serialize for Box<T> {
-        #[inline(always)]
-        fn tls_serialize<W: Write>(&self, writer: &mut W) -> Result<usize, Error> {
-            self.as_ref().tls_serialize(writer)
-        }
+impl<T: Serialize> Serialize for Box<T> {
+    #[cfg(feature = "std")]
+    #[inline(always)]
+    fn tls_serialize<W: Write>(&self, writer: &mut W) -> Result<usize, Error> {
+        self.as_ref().tls_serialize(writer)
     }
+}
 
-    impl<T: SerializeBytes> SerializeBytes for Box<T> {
-        #[inline(always)]
-        fn tls_serialize(&self) -> Result<Vec<u8>, Error> {
-            self.as_ref().tls_serialize()
-        }
+impl<T: SerializeBytes> SerializeBytes for Box<T> {
+    #[inline(always)]
+    fn tls_serialize(&self) -> Result<Vec<u8>, Error> {
+        self.as_ref().tls_serialize()
     }
+}
 
-    impl<T: Deserialize> Deserialize for Box<T> {
-        #[inline(always)]
-        fn tls_deserialize<R: Read>(bytes: &mut R) -> Result<Self, Error> {
-            T::tls_deserialize(bytes).map(Box::new)
-        }
+impl<T: Deserialize> Deserialize for Box<T> {
+    #[cfg(feature = "std")]
```

Comment:
Similar concern to [this comment](https://github.com/RustCrypto/formats/pull/1586/files/619bf0ba168739d80006824aa837f4d361f8fafc..1500766531dabcbcf071da4173fdf0fee2587d27#r1819330935)